### PR TITLE
Adds a mounted plasma cutter for Sabres

### PR DESCRIPTION
**Pull request by benbot16**
## About The Pull Request

This adds a new utility mount for Sabre, a mounted plasma cutter. This is essentially the same as a mech's heavy plasma cutter, and it draws from the Sabre's battery to fire.

This also adds the necessary code to allow a fighter to use its weapons on asteroid Z-levels, and the necessary code to allow a fighter weapon to fire laterally.

The Sabre's TAC console is still a bit fucky with this, as it is for most small-craft weapons, but that'd probably need to get fixed somewhere else so it doesn't conflict with stuff.

## Why It's Good For The Game

Someone suggested this, and I figure that more NSV-specific tools for miners is a good thing, especially since the vanilla counterpart to this (using mechs) isn't very effective on NSV's zero-g asteroids.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

https://user-images.githubusercontent.com/9423435/213271181-ac8718f0-9ce9-4c25-bb52-0a835b26513f.mp4

</details>

## Changelog
:cl:
add: Adds a sabre-mounted plasma cutter. This can be printed at cargo protolathes and uses ship power to fire.
/:cl:

### CODE REVIEW

**Comment by Karmic-Skink** [comment]
GG doesn't even know what the ship class is called.

**Comment by benbot16** [comment]
> GG doesn't even know what the ship class is called.

This is why sleep is a good thinf

File: nsv13/code/modules/overmap/fighters/modules/utility_modules.dm [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
+	slot = HARDPOINT_SLOT_UTILITY_PRIMARY
+	var/projectile = /obj/item/projectile/plasma/adv/sabre
+	var/charge_to_fire = 2500
+
+/obj/item/fighter_component/primary/plasmacutter/get_ammo()
+	var/obj/structure/overmap/small_craft/F = loc
+	if(!istype(F))
+		return FALSE
+	var/obj/item/fighter_component/battery/B = F.loadout.get_slot(HARDPOINT_SLOT_BATTERY)
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/overmap/fighters/modules/utility_modules.dm:24]: Using plasma as ammo would be more interesting imo, but we can use this for now.

File: nsv13/code/modules/overmap/overmap.dm [hunk 2]
```diff
@@ -823,7 +823,8 @@ Proc to spool up a new Z-level for a player ship and assign it a treadmill.
 		M.Knockdown(2 SECONDS)
 
 /obj/structure/overmap/proc/can_change_safeties()
-	return (SSmapping.level_trait(loc.z, ZTRAIT_OVERMAP))
+	// Safeties can be toggled on the overmap or on small dockable maps like ruins and asteroids
+	return (SSmapping.level_trait(loc.z, ZTRAIT_OVERMAP) || SSmapping.level_trait(loc.z, ZTRAIT_RESERVED))
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/overmap/overmap.dm:827]: Unrelated, but we should make a better way to detect asteroids.

**Review by Bokkiewokkie — APPROVED** [review]

**Review by Bokkiewokkie — APPROVED** [review]

**Review by covertcorvid — APPROVED** [review]

**Pull request merged**